moveit_ros/moveit_servo/include/moveit_servo/servo_parameters.cpp

```diff
@@ -0,0 +1,278 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, PickNik LLC
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik LLC nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author    : Adam Pettinger
+   Desc      : Declares, loads, and checks ServoParameters for Servo
+   Title     : servo_parameters.cpp
+   Project   : moveit_servo
+   Created   : 07/02/2020
+*/
+
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <type_traits>
+
+#include <moveit_servo/servo_parameters.h>
+
+namespace moveit_servo
+{
+template <typename T>
+bool declareAndGetParam(const std::string& param_name, T& output_value, rclcpp::Node::SharedPtr& node,
```

Comment:
I really like what you did here.  The only thing that I see is that the output parameter is in the middle of the parameter list.  I think it would be clearer if you either made it the first or last parameter.

Next, you aren't changing the shared_ptr of node in this so it should be a `const shared_ptr<rclcpp::Node>&`.

I really like what you did with the double/int issue and this whole function for making the parameter declaration and getting simpler.

Reply:
Done!